python_ta/checkers/possibly_undefined_checker.py

```diff
@@ -0,0 +1,158 @@
+"""checker for variables that might not be defined in the program.
+"""
+from typing import Union
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from python_ta.cfg.graph import CFGBlock
+from astroid.node_classes import NodeNG
+from typing import Set, List
+
+
+class PossiblyUndefinedChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+    # name is the same as file name but without _checker part
+    name = 'possibly_undefined'
+    # use dashes for connecting words in message symbol
+    msgs = {'E9969': ('This variable might not be defined when this statement is executed.',
```

Comment:
What message-id should it be? @david-yz-liu

Reply:
`E9969` is okay with me, we don't currently have a systematic way of naming new errors.